src/Microsoft.ML.Data/DataView/ArrayDataViewBuilder.cs

```diff
@@ -211,20 +212,20 @@ public DataView(IHostEnvironment env, ArrayDataViewBuilder builder, int rowCount
                 _host.Assert(builder._names.Count == builder._columns.Count);
                 _columns = builder._columns.ToArray();
 
-                var schemaCols = new Schema.Column[_columns.Length];
-                for(int i=0; i<schemaCols.Length; i++)
+                var schemaBuilder = new SchemaBuilder();
+                for(int i=0; i< _columns.Length; i++)
                 {
-                    var meta = new Schema.Metadata.Builder();
+                    var meta = new MetadataBuilder();
```

Comment:
>var meta = new MetadataBuilder(); [](start = 20, length = 33)

Out of curiosity.
In the past metadata builder was part of schema, and that indicated it relationship with schema and what metadata pointless outside of schema. 
Why we separate metadata from schema?

Reply:
One of the feedbacks from the API review with .NET is that nested classes should be avoided, if possible. `Schema.Column`, and `Schema.Metadata` are acceptable exceptions (since these objects are nested inside `Schema`), but not builders.

Also, we want to keep the option to move builders into a separate namespace, distinct from `Microsoft.ML.Data`

---
In reply to: [235485750](https://github.com/dotnet/machinelearning/pull/1680#discussion_r235485750) [](ancestors = 235485750)